test/TesterInternal/EventStoreTests/CommonEventStoreTests.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TestExtensions;
+using Xunit.Abstractions;
+using Orleans.Runtime.Storage;
+using Orleans.Storage;
+using Orleans.Runtime;
+using Xunit;
+using Orleans;
+using System.Threading;
+
+namespace Tester.EventStorageTests
+{
+    /// <summary>
+    /// Common tests for event stores.
+    /// To use these tests, derive from this class, and specify 
+    /// the store to test via the abstract property.
+    /// All tests create fresh GUIDs, so no cleaning is necessary in between tests.
+    /// </summary>
+    public abstract class CommonEventStoreTests
```

Comment:
@jdom I told @sebastianburckhardt that I would submit a PR with the test separation to this PR's branch. Probably doing tomorrow.